sql/plan/show_grants.go

```diff
@@ -114,6 +114,24 @@ func (n *ShowGrants) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedO
 	}
 }
 
+// generatePrivStrings gathers all the privileges and if with grant option should be printed
+func generatePrivStrings(privs []sql.PrivilegeType) (string, string) {
```

Comment:
Seems strange to return `" WITH GRANT OPTION"` as a separate string, makes more sense to just append it to the end and return a single string since you're just appending it at the call site anyway

Reply:
This helper methods returns just the privilege part (`SELECT, INSERT, ...`), and `WITH GRANT OPTION` shows up later.
The format of the string changes depending on what level the privilege is printing (global vs database vs table).

Maybe the best way is to have the helper method also determine the format?